macOS/DuckDuckGo/Feedback/Model/Feedback.swift

```diff
@@ -35,5 +35,80 @@ struct Feedback {
     let comment: String
     let appVersion: String
     let osVersion: String
+    let subcategory: String
 
+    init(category: Category,
+         comment: String,
+         appVersion: String,
+         osVersion: String,
+         subcategory: String = "") {
+        self.category = category
+        self.comment = comment
+        self.appVersion = appVersion
+        self.osVersion = osVersion
+        self.subcategory = subcategory
+    }
+}
+
+extension Feedback {
+
+    static func from(selectedPills: [String],
+                     text: String,
+                     appVersion: String,
+                     category: Feedback.Category,
+                     problemCategory: ProblemCategory?) -> Feedback {
+        let selectedOptionsString = selectedPills.map { $0.toTag }.joined(separator: ",")
+        let description = text.isEmpty ? category.toString : text
+        var subcategory = ""
+
+        if let problemCategory = problemCategory {
+            subcategory = "\(problemCategory.name.toTag),\(selectedOptionsString)"
+        } else {
+            subcategory = "\(selectedOptionsString)"
+        }
+
+        return Feedback(category: category,
+                        comment: description,
+                        appVersion: appVersion,
+                        osVersion: "\(ProcessInfo.processInfo.operatingSystemVersion)",
+                        subcategory: subcategory)
+    }
+}
+
+extension String {
+    var toTag: String {
+        return self
+            .trimmingCharacters(in: .whitespacesAndNewlines)
+            .lowercased()
+            .replacingOccurrences(of: "'", with: "")
+            .replacingOccurrences(of: "&", with: "-and-")
+            .replacingOccurrences(of: "/", with: "-")
+            .replacingOccurrences(of: ".", with: "-")
+            .replacingOccurrences(of: "(", with: "")
+            .replacingOccurrences(of: ")", with: "")
+            .replacingOccurrences(of: "\"", with: "")
+            .replacingOccurrences(of: "!", with: "")
+            .replacingOccurrences(of: "?", with: "")
+            .replacingOccurrences(of: "%", with: "-percent-")
+            .replacingOccurrences(of: "@", with: "-at-")
+            .replacingOccurrences(of: "#", with: "-number-")
+            .replacingOccurrences(of: "$", with: "-dollar-")
+            .replacingOccurrences(of: "\\s+", with: "-", options: .regularExpression)
+            .replacingOccurrences(of: "[^a-z0-9-]", with: "", options: .regularExpression)
+            .replacingOccurrences(of: "-+", with: "-", options: .regularExpression)
+            .trimmingCharacters(in: CharacterSet(charactersIn: "-"))
+    }
+}
+
+extension Feedback.Category {
```

Comment:
The legacy form was only text, so the backend expects it. In the new forms, the text is optional, so if the user doesn’t add any text, we send the placeholders used here.